### PR TITLE
feat(invitations): Expand invitation context to external metadata

### PR DESCRIPTION
**Pull request by javirln**
This pull request introduces changes to improve the handling of organization invitations, particularly in the context of joining groups and projects. The updates include modifying the `OrgInvitationContext` structure to use pointers for UUID fields, adding support for external metadata, and enhancing test coverage to handle edge cases like nil UUIDs.

Before, since the groupID and projectID were not pointers, they were being stored in the DB as their nil representation of UUUID the `0000-0000...` by adding a pointer, the are not store at all. The UUID != uuid.Nil check is there for already stored data.

### CODE REVIEW

File: app/controlplane/pkg/biz/orginvitation.go [hunk 1]
```diff
@@ -338,20 +341,20 @@ func (uc *OrgInvitationUseCase) FindByID(ctx context.Context, invitationID strin
 // processGroupMembership adds a user to a group if the invitation context contains a group to join
 func (uc *OrgInvitationUseCase) processGroupMembership(ctx context.Context, invitation *OrgInvitation, orgUUID uuid.UUID, userUUID uuid.UUID) error {
 	// Skip if there's no group to join in the invitation context
-	if invitation.Context == nil || invitation.Context.GroupIDToJoin == uuid.Nil {
+	if invitation.Context == nil || invitation.Context.GroupIDToJoin == nil || *invitation.Context.GroupIDToJoin == uuid.Nil {
```
Comment (jiparis) [inline app/controlplane/pkg/biz/orginvitation.go:344]: much better!

**Review by jiparis — APPROVED** [review]
